src/main.rs

```diff
@@ -50,6 +50,41 @@ fn handle_addition(settings: &Settings) {
 
 fn handle_search(settings: &Settings) {
     let history = History::load(settings.history_format);
+
+    if settings.no_ui {
+        history.build_cache_table(
+            &settings.dir.to_owned(),
+            &Some(settings.session_id.to_owned()),
+            None,
+            None,
+            None,
+            settings.limit.to_owned(),
+        );
+        let matches = history.find_matches(
+            &settings.command,
+            settings.results as i16,
+            settings.fuzzy,
+            &mcfly::settings::ResultSort::Rank,
+        );
+
+        let cmd = match matches.first() {
+            Some(cmd) => cmd.cmd.to_owned(),
+            None => settings.command.to_owned(),
+        };
+        let cmd_t = cmd.trim();
+        if cmd_t.is_empty() {
+            return;
+        }
+
+        let out = format!("mode display\ncommandline {}\n", cmd_t);
+        if let Some(path) = &settings.output_selection {
```

Comment:
Instead of requiring `--output_selection`, you could allow two options:

1. With `--output_selection`, write to the `--output_selection` selection file as you're doing now.
2. Without output_selection, instead just print one or more matches to STDOUT for scripts to parse.

Reply:
It was originally written this way, however I found that we are happily `print!`-ing many messages to STDOUT, which (given the current output format) might mess up parsing for simple scripts.

Reply:
We should only be using print! to output when there's an error. Those should probably be converted to use STDERR.

Reply:
We should update all `print!` to `eprint(ln)!` before merging this.

Reply:
Agreed!